codechecker_lib/analyzers/result_handler_base.py

```diff
@@ -180,37 +175,36 @@ def analyzed_source_file(self, file_path):
 
     def get_analyzer_result_file(self):
         """
-        file where the analyzer should put the analyzer result
-        extra_info should make the result file uniqe
-        each result handler should collect the provided analyzer_result files
-        for later cleanup
+        generate a result filename where the analyzer should put the analyzer result
+        result file should be removed by the result handler if not required anymore
         """
-        analyzed_file = self.analyzed_source_file
-        _, analyzed_file_name = ntpath.split(analyzed_file)
+        if not self.__res_file:
+            analyzed_file = self.analyzed_source_file
+            _, analyzed_file_name = ntpath.split(analyzed_file)
 
-        out_file_name = str(self.buildaction.id) + '_' + \
-                         str(self.buildaction.analyzer_type) \
-                        + '_' + analyzed_file_name + '.out'
+            uid = str(uuid.uuid1()).split('-')[0]
 
-        out_file = os.path.join(self.__workspace, out_file_name)
-        self.add_result_file(analyzed_file_name, out_file)
+            out_file_name = str(self.buildaction.analyzer_type) + \
+                '_' + analyzed_file_name + '_' + uid + '.out'
```

Comment:
Is .out a good extension? In case these files are plist, I would prefer to keep the plist extension.